packages/web3-providers-ws/src/deferredPromise.ts

```diff
@@ -0,0 +1,35 @@
+export class DeferredPromise<T> {
+	public realPromise: Promise<T>;
+
+	private resolveFunction!: (value: T | PromiseLike<T>) => void;
+	private rejectFunction!: (reason?: unknown) => void;
```

Comment:
I prefer using `_`, it categorizes the properties the IDE suggests when you start typing: `this.`

Reply:
@jdevcs As the document you referred says. 

> This is NOT a prescriptive guideline for the TypeScript community

And the majority of the Typescript community tend to follow the standard of having `_` prefix for the `private` and `protected`  attributes and functions. And the reason is the value addition to the readability. While writing the code first time you may not notice a difference. But when you will be reading the same code after few months, you will notice that just knowing by convention which is not public adds a lot of context to the code line. 


The other major reason why Typescript itself suggest to not use `_` prefix is the availability of [native private properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) in Javascript and was introduced in [Typescript 3.8](https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/#ecmascript-private-fields). But that feature is yet on Stage 3, where further discussion and refinement is needed. 

So until we have that native private fields feature adopted, we should stick to using `_` as prefix for private and protected fields.

Reply:
Its not good convention to use (manual) naming prefix or post-fix for differentiating variables access modifier traits.
[Google guidelines](https://google.github.io/styleguide/tsguide.html#naming-style)

Reply:
I had explained it very briefly already. The reason to not use prefix just because of some published guide is not enough. There are two questions to answer: 

1. Does adding a prefix to private variables add some value? 
2. Does adding a prefix to private variables create some issue? 

The answer to first question is Yes, it add a huge value towards code readability. Reading code from color coding of the IDE is not effective (no one had such brain power). Reading code with a known convention is most effective pattern. 

The answer to second question is No. Adding prefix does not create any issue in the code. Neither of the guides provide any reason or problem this convention may cause. 

So with that said we should just focus on the value addition and follow same pattern in the code. In addition myself and @spacesailor24 agrees of adding prefix for the private/protected fields and functions of the classes, so if you follow it we will have consistent code in the library.